samples/kafka-avro/src/main/java/com/example/kafka/KafkaAvroApplication.java

```diff
@@ -35,17 +35,17 @@ public NewTopic topic() {
 	}
 
 	@Bean
-	public ConcurrentMessageListenerContainer<String, Thing3> manualListenerContainer(MyMessageListener listener,
-			ConcurrentKafkaListenerContainerFactory<String, Thing3> factory) {
+	public ConcurrentMessageListenerContainer<Object, Object> manualListenerContainer(MyMessageListener listener,
```

Comment:
I think we should not change the sample like that but instead fix the AOT processing to handle this use case cc @snicoll.

Reply:
Unfortunately, this isn't that easy. There's no `ConcurrentMessageListenerContainer<String, Thing3>` bean in the context so, technically, the current code is wrong. It is only wrong because Framework has several layers of fallback before throwing an exception that we don't have. 

I discussed this with @jhoeller and he was more keen to fix this rather than relying on the fallback. What bothers me is the fact we can't enforce that with the regular JVM/runtime so it always feel like a limitation of the AOT engine....

Reply:
This is rather unfortunate; users have relied on this capability for years.

Reply:
I agree. Feel free to create a separate issue and we'll see what we can do.